packages/app-elements/src/providers/GTMProvider.tsx

```diff
@@ -0,0 +1,60 @@
+import isEmpty from 'lodash/isEmpty'
+import { createContext, useContext, useEffect, useMemo } from 'react'
+import TagManager, { type DataLayerArgs } from 'react-gtm-module'
+
+interface GTMProviderValue {
+  push: (
+    data: DataLayerArgs['dataLayer'],
+    customDataLayerName?: DataLayerArgs['dataLayerName']
+  ) => void
+}
+
+const GTMContext = createContext<GTMProviderValue | null>(null)
+
+/**
+ * Hook that is aimed to expose the provider context containing the `tagManager` object and its methods.
+ */
+export const useTagManager = (): GTMProviderValue | null => {
+  return useContext(GTMContext)
+}
+
+interface GTMProviderProps {
+  /**
+   * Entire app content
+   */
+  children: React.ReactNode
+  /**
+   * Optional Google Tag Manager id. If set the provider will take care of GTM initialization.
+   */
+  gtmId?: string
```

Comment:
Are we going to use a stricter `string` type here? maybe with a regex?

Reply:
Good idea! I just added a conditional based regex check to permit the initialization..